Cdms.Business/Pipelines/TerminatePipelineBase.cs

```diff
@@ -0,0 +1,18 @@
+using Cdms.Business.Pipelines.Matching;
+using MediatR;
+
+namespace Cdms.Business.Pipelines;
+
+public abstract class TerminatePipelineBase<TRequest, TResult> : IPipelineBehavior<TRequest, TResult>
+    where TRequest : notnull
+    where TResult : PipelineResult
+{
+    public async Task<TResult> Handle(
+        TRequest request,
+        RequestHandlerDelegate<TResult> next,
+        CancellationToken cancellationToken)
+    {
+        var result = (TResult)Activator.CreateInstance(typeof(TResult), false)!;
```

Comment:
Not 100% sure about this pipeline step, but as long as TResult has a default constructor, then you can call "new TResult()" instead of using reflection